babelsberg/constraintinterpreter.js

```diff
@@ -1226,6 +1330,37 @@ Object.subclass('ConstrainedVariable', {
             this._resetIsSolveable();
         }
     },
+
+    /**
+     * Removes all external variables which belong only to abandoned constraints.
+     */
+    compactExternalVariables: function() {
+        var externalVariable;
+        var wasAbandonedOrUsesOtherSolver = function(constraint) {
+            return constraint.solver !== externalVariable.__solver__ ||
+                constraint.isAbandoned;
+        };
+
+        var externalVariableKeysToRemove = [];
+        for (var solverUUID in this._externalVariables) {
+            if (this._externalVariables[solverUUID] === null) {
+                externalVariableKeysToRemove.push(solverUUID);
+                continue;
+            }
+            if (!this._externalVariables[solverUUID].isConstraintObject)
+                continue;
+            externalVariable = this._externalVariables[solverUUID];
+            // search enabled Constraint for this variable's solver
+            var allAbandoned = this._constraints.every(wasAbandonedOrUsesOtherSolver);
```

Comment:
* typo: search abandonedConstraints
* how does a "uses other solver" scenario exist? `this.solver` should be set in the constructor and then be left untouched (right?)